### ISSUE TITLE
Medical Transformer: Universal Brain Encoder for 3D MRI Analysis

### ISSUE THREAD

**Issue by ghuijo** [yskim0/One-Day-One-Paper#10]
## Abstract
 * 새로운 transfer learning framework인 Medical Transformer를 제안한다.
 * Medical Transformer는 효과적으로 3D volumetric images를 2D image slices의 형태로 modeling할 수 있다.
 * proxy task로써 masked encoding vector prediction을 위해 model을 self-supervised learning manner로 pre-train하고, 이때 large-scale의 MRI dataset을 사용한다.
* pre-train된 모델은 3가지 downstram tasks로 평가한다.
  > brain disease diagnosis
  > brain age prediction
  > brain tumor segmentation
* classification과 regression tasks의 parameters 수를 92%, segmentation은 97%를 줄이고도 성능이 우수하다.

## Related Work
* Transformers in Vision Models
  * Vision Transformer (ViT)
* Fully supervised Pre-training
  * Inflated 3D (I3D)
  * NiftyNet
* Self-supervised Pre-training
  * Model Genesis

## Proposed Method
**_large-scale의 3D brain MRI dataset으로 pre-train시킨 백본 네트워크를 downstream task인 3D MRI analyses로 transfer시키는 것._**
***
![image](https://user-images.githubusercontent.com/67627471/127733438-44fd6ff6-4705-42ce-a401-556ec0ed83db.png)
***
1. large-scale의 3D brain MRI dataset을 만듦.
    > Information extraction from Images (IXI)
     Cambridge Centre for Ageing and Neuroscience (Cam-CAN)
     Autism Brain Imaging Data Exchange (ABIDE)
2. data variation 문제가 있으므로, spatial and intensity distribution normalization 수행.
3. 백본 네트워크의 convolutional encoder와 transformer, prediction netwrork를 기반으로 3D volumetric 이미지를 3개의 planes으로 2D slices로 나눔.
    > 각각의 plane은 model parameters를 공유하지 않음.
4. spatial features 위해 positional encoding / odering slices 위해 sement encoding 거침.
5. 나눈 slices를 네트워크에 input.
6. transformer 지난 후 3D volume feature map 생성.
7. prediction network이 final task-specific prediction을 output.
***
* transfer learning을 3단계로 진행
    > pre-training convolutional encoder with an exemplar network
    > pre-training transformer for a masked encoding vector prediction as our proxy task of a self-supervised learning (SSL)
    > fine-tuning the backbone and prediction network for downstream medical tasks

* encoder pre-training 과정
<image src="https://user-images.githubusercontent.com/67627471/127733655-c800a6ec-91cd-4d81-ab53-d6db79eceee5.png" width="60%" height="60%"/>

* fine-tuning 과정
<image src="https://user-images.githubusercontent.com/67627471/127733686-2c256a33-bf48-4341-b9e6-a9805df4a415.png" width="90%" height="90%"/>

***

* multi-scale / single scale 각각의 결과를 prediction network 전에 어떻게 합치는 지 이해가 안 됨.
* 전처리 과정이 긴듯.

**Comment by yskim0** [comment]
우와 저는 특정 도메인없이 general 한 것만 보다보니 CT에 적용되는 게 되게 신기하고 좋아요!!!!

**Comment by ghuijo** [comment]
@yskim0 transformer는 그 자체로 활용성이 무궁무진한 model인 것 같아요. 최근에 medical 분야에서 논문이 꽤 나오고 있어서 재밌어요!
